plugins/janus_audiobridge.c

```diff
@@ -1503,17 +1535,35 @@ void janus_audiobridge_incoming_rtp(janus_plugin_session *handle, int video, cha
 		}
 		/* Enqueue the decoded frame */
 		janus_mutex_lock(&participant->qmutex);
-		g_queue_push_tail(participant->inbuf, pkt);
-		/* Keep the queue at max of 150 packets, though */
-		if(g_queue_get_length(participant->inbuf) > 150) {
-			/* Remove the oldest packet */
-			janus_audiobridge_rtp_relay_packet *pkt = g_queue_pop_head(participant->inbuf);
-			if(pkt != NULL) {
-				if(pkt->data)
-					g_free(pkt->data);
-				pkt->data = NULL;
-				g_free(pkt);
-				pkt = NULL;
+		/* Insert packets sorting by sequence number */
+		participant->inbuf = g_list_insert_sorted(participant->inbuf, pkt, &janus_audiobridge_rtp_sort);
+		if(participant->prebuffering) {
+			/* Still pre-buffering: do we have enough packets now? */
+			if(g_list_length(participant->inbuf) == DEFAULT_PREBUFFERING) {
+				participant->prebuffering = FALSE;
+				JANUS_LOG(LOG_VERB, "Prebuffering done! Finally adding the user to the mix\n");
+			} else {
+				JANUS_LOG(LOG_VERB, "Still prebuffering (got %d packets), not adding the user to the mix yet\n", g_list_length(participant->inbuf));
+			}
+		} else {
+			/* Make sure we're not queueing too many packets: if so, get rid of the older ones */
+			if(g_list_length(participant->inbuf) >= DEFAULT_PREBUFFERING*2) {
```

Comment:
It was just an attempt to put a cap on the buffer queue, which had no limit before. ```DEFAULT_PREBUFFERING``` being 6 packets, this means the queue contains 12 packets tops, hence about 240ms of audio (each RTP packet is 20ms). You may want to increase that if in your experience you noticed this causes trouble, and if you're ok with a potentially increased delay for affected users: e.g., ```3*6=18``` would mean up to 360ms of delay for users who get their queue filled. Users without issues, instead, will only have about 100ms of delay if you don't change the value of ```DEFAULT_PREBUFFERING``` itself.

If you come up with some values that give a good experience in most of your scenarios, just let us know and we'll make those the defaults.